src/App.tsx

```diff
@@ -1,14 +1,81 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
+import { TodoItem } from './components/Todo';
+
+type Sort = 'All' | 'Active' | 'Completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortBy, setSortBy] = useState<Sort>('All');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  if (!errorMessage) {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
+
+  const handleCompltete = (id: number) => {
+    const completedTodo = todos.map(todo => {
+      if (todo.id === id) {
+        setLoadingIds([...loadingIds, todo.id]);
+
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      setTimeout(() => {
+        const loading = loadingIds.filter(item => {
+          return item !== todo.id;
+        });
+
+        setLoadingIds(loading);
+      }, 500);
+
+      return todo;
+    });
+
+    setTodos(completedTodo);
+  };
+
+  const sortTodos = () => {
+    const filteredTodos = [...todos].filter(todo => {
+      if (sortBy === 'Active') {
+        return todo.completed === false;
+      }
+
+      if (sortBy === 'Completed') {
+        return todo.completed === true;
+      }
+
+      return todo;
+    });
+
+    return filteredTodos;
+  };
+
+  const filteredTodos = sortTodos();
```

Comment:
If the constant is named filteredTodos I recommend naming function such as getFileteredTodos or something like that.